loadsave.c

```diff
@@ -169,7 +169,15 @@ LOAD()
 			// IO addresses
 		} else if(start < 0xc000) {
 			// banked RAM
-			bytes_read = fread(RAM + ((uint16_t)memory_get_ram_bank() << 13) + start, 1, 0xc000 - start, f);
+			while(1) {
+				size_t len = 0xc000 - start;
+				bytes_read = fread(RAM + ((uint16_t)memory_get_ram_bank() << 13) + start, 1, len, f);
+				if(bytes_read < len) break;
+
+				// Wrap into the next bank
+				start = 0xa000;
+				write6502(0x9f61, 1 + read6502(0x9f61));
```

Comment:
Better use `memory_set_ram_bank()`/`memory_get_ram_bank()`, which are already exported.

Reply:
The problem is that those functions don't update `via1registers`.  I believe I could use `via1_write()` instead.  Thoughts?

Reply:
I cleaned up this part in the VIA in e805d14. Would it work now?

Reply:
Much better.  Would you like me to rebase on the current master and crush into one commit?